modules/geo-layers/src/tile-layer/utils.js

```diff
@@ -29,17 +30,25 @@ export const urlType = {
   }
 };
 
-export function getURLFromTemplate(template, properties) {
+export function getURLFromTemplate(template, properties, indexing) {
   if (!template || !template.length) {
     return null;
   }
   if (Array.isArray(template)) {
     const index = Math.abs(properties.x + properties.y) % template.length;
     template = template[index];
   }
-  return template.replace
-    ? template.replace(/\{ *([\w_-]+) *\}/g, (_, property) => properties[property])
-    : null;
+
+  let {x, y, z} = properties;
+  if (indexing === 'tms') {
+    y = Math.pow(2, z) - y - 1;
+  }
+
+  return template
+    .replace('{x}', x)
+    .replace('{y}', y)
+    .replace('{z}', z)
+    .replace('{-y}', Math.pow(2, z) - y - 1);
```

Comment:
> either by passing the indexing: 'tms' prop or by using {-y} in the data prop

What should `-y` be, the OSM y or TMS y?

According to your implementation, the effect of `indexing: 'tms'` and `{-y}` negate each other. Can they be both used? Avoid overwriting `y` in this function for better clarity.

Reply:
I struggled with this while first drawing up the PR last week. Elsewhere I've seen support for both 1) setting a TMS: true prop and 2) having `{-y}` refer to a TMS coordinate.

Reply:
Is this correct though? If an user specified `indexingScheme` then they would expect `tile.y` to be in the TMS system, which it is not. Since we do not overwrite `tile.y`, `-y` and no `indexingScheme` may be the less confusing solution?

Reply:
We discussed this in https://github.com/visgl/deck.gl/issues/4757#issuecomment-655721368

> True that the user will need to understand that `y` is not from the TMS indexing system if they want to handle this information programmatically. However, as far as I can tell, the original issue is about the automatic URL resolution

I think part of my original intention with overwriting `y` was so that both the `data` url and `x, y, z` would be in `indexingScheme`. I think `indexingScheme` is a little more flexible than `{-y}`, so if I had to choose one or the other I'd choose the former.

Reply:
That is also fine, though you'll have to move the `indexingScheme` handling to `Tileset2D` so that all tile indices are correctly populated.

Reply:
My comment that you quoted were referring to using `{-y}` instead of `indexingScheme`